vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/communication/IOThread.java

```diff
@@ -339,11 +345,14 @@ private ThreadState cycle(ThreadState threadState) {
                     gatewayThrottler.handleCall(processResponse.transitiveErrorCount);
                 }
                 catch (ServerResponseException ser) {
-                    log.info("Problems while handing data over to gateway " + endpoint + ": " + Exceptions.toMessageString(ser));
+                    log.log(Level.INFO, "Problems while handing data over to endpoint '" + endpoint
+                            + "'. Will re-try. Endpoint responded with a HTTP response code we did not expect. '"
```

Comment:
Replace "Endpoint responded with a HTTP response code we did not expect" with "Unexpected HTTP response code from endpoint" - there is no we ...